websites/D/DEX Screener/presence.ts

```diff
@@ -0,0 +1,201 @@
+interface DexTokenInfo {
+	name: string;
+	symbol: string;
+	price: number;
+	chainId: string;
+	thumbnail: string;
+	chainSymbol: string;
+}
+
+const tokenCache: Map<string, DexTokenInfo> = new Map();
+
+async function getTokenInfo(
+	tokenAddress: string,
+	chainId: string
+): Promise<DexTokenInfo | null> {
+	const cacheKey = `${chainId}-${tokenAddress}`;
+	if (tokenCache.has(cacheKey)) return tokenCache.get(cacheKey) || null;
+
+	const fetchTokenInfo = async () => {
+		let response = await fetch(
+				`https://api.dexscreener.com/latest/dex/pairs/${chainId}/${tokenAddress}`
+			),
+			data = await response.json();
+
+		if (!data.pairs || data.pairs.length === 0) {
+			response = await fetch(
+				`https://api.dexscreener.com/latest/dex/tokens/${tokenAddress}`
+			);
+			data = await response.json();
+		}
+
+		if (data && data.pairs && data.pairs.length > 0) {
+			const pair = data.pairs[0],
+				getTitlePrice = () => {
+					const priceMatch = document.title.match(/\$([0-9.]+)/);
+					return priceMatch ? parseFloat(priceMatch[1]) : 0;
+				};
+			let price = getTitlePrice();
+			// price get from title instead of API, for efficiency
+			const tokenInfo = {
+				name: pair.baseToken.name,
+				symbol: pair.baseToken.symbol,
+				price,
+				chainId: pair.chainId,
+				thumbnail: pair.info.imageUrl,
+				chainSymbol: pair.quoteToken.symbol,
+			};
+			tokenCache.set(cacheKey, tokenInfo);
+
+			// Update price periodically based on title changes
+			setInterval(() => {
+				const newPrice = getTitlePrice();
+				if (newPrice !== price) {
+					tokenInfo.price = newPrice;
+					price = newPrice;
+				}
+			}, 10000);
+			return tokenInfo;
+		}
+		return null;
+	};
+	return await fetchTokenInfo();
+}
+
+function capitalize(string: string): string {
+	return string.charAt(0).toUpperCase() + string.slice(1);
+}
+// end function
+
+// PRESENCEDECLARATION
+const presence = new Presence({
+		clientId: "1326168769403289683",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/LWNN5i1.jpeg",
+}
+// end PRESENCEDECLARATION
+const chainLogo = {
+	// i added this assets because these logo is not in .png while get from website
+	ethereum: "https://i.imgur.com/u2u5w93.jpeg",
+	bsc: "https://i.imgur.com/Eaa2uog.jpeg",
+	solana: "https://i.imgur.com/RUJenzh.jpeg",
+	base: "https://i.imgur.com/yMUaMZq.jpeg",
+	sui: "https://i.imgur.com/1U8Kuh0.jpeg",
+	aptos: "https://i.imgur.com/M2PnsZo.jpeg",
+	arbitrum: "https://i.imgur.com/C8y29U6.jpeg",
+	avalanche: "https://i.imgur.com/yApPGyO.jpeg",
+	cronos: "https://i.imgur.com/K9HSwqr.jpeg",
+	fantom: "https://i.imgur.com/2h7g96v.jpeg",
+	hyperliquid: "https://i.imgur.com/1YmOCAI.jpeg",
+	ink: "https://i.imgur.com/tvhGF3x.jpeg",
+	linea: "https://i.imgur.com/4kxwQYb.jpeg",
+	optimism: "https://i.imgur.com/NkMIdoE.jpeg",
+	osmosis: "https://i.imgur.com/7XyInA2.jpeg",
+	polygon: "https://i.imgur.com/OAj35XY.jpeg",
+	zksync: "https://i.imgur.com/su2f6QM.jpeg",
```

Comment:
This took way too long
```suggestion
	ethereum: "https://i.imgur.com/sRQsRgt.png",
	bsc: "https://i.imgur.com/P3R5xml.png",
	solana: "https://i.imgur.com/yzW1XmK.png",
	base: "https://i.imgur.com/RFCn600.png",
	sui: "https://i.imgur.com/nRLlRS3.png",
	aptos: "https://i.imgur.com/HnRC0zH.png",
	arbitrum: "https://i.imgur.com/62omOHV.png",
	avalanche: "https://i.imgur.com/EnQN7bJ.png",
	cronos: "https://i.imgur.com/X884cZX.png",
	fantom: "https://i.imgur.com/nBuhY97.png",
	hyperliquid: "https://i.imgur.com/4YsBubJ.png",
	ink: "https://i.imgur.com/pqBMg7y.png",
	linea: "https://i.imgur.com/9l7YKgI.png",
	optimism: "https://i.imgur.com/8c49Wtl.png",
	osmosis: "https://i.imgur.com/aPdpP43.png",
	polygon: "https://i.imgur.com/GLDY9Z0.png",
	zksync: "https://i.imgur.com/5bK6sx8.png",
```

Reply:
thank you!